### PR TITLE
Configure email notifications

### PR DESCRIPTION
**Pull request by maxim12321**

### CODE REVIEW

**Review by Darina007 — APPROVED** [review]

File: src/main/java/com/team4/testingsystem/handlers/EmailNotificationEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        this.templateResolver = templateResolver;
+        this.mailSender = mailSender;
+    }
+
+    @EventListener
+    public void onApplicationEvent(NotificationDTO notificationDTO) throws MessagingException {
+        MimeMessage mimeMessage = mailSender.createMimeMessage();
+
+        MimeMessageHelper message = new MimeMessageHelper(mimeMessage, true, "UTF-8");
+        message.setFrom("untitled.noreply@gmail.com");
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/handlers/EmailNotificationEventHandler.java]: I think, from mail can be extracted to properties

File: src/main/java/com/team4/testingsystem/handlers/EmailNotificationEventHandler.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        message.setTo(notificationDTO.getUserEmail());
+        message.setSubject("Untitled testing system");
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/handlers/EmailNotificationEventHandler.java]: What do you think about setting a subject based on the event?

File: src/main/resources/application-release.properties [hunk 3]
```diff
@@ -8,3 +8,12 @@ cloud.aws.credentials.access-key=${AWS_ACCESS_KEY}
...
 cloud.s3.bucket-name=untitled-testing-system
+
+# Email notifications
+email-notifications.enabled=true
+email-notifications.template-path.test-assigned=test_assigned.ftlh
+email-notifications.template-path.test-deassigned=test_assigned.ftlh
+email-notifications.template-path.test-started=test_assigned.ftlh
+email-notifications.template-path.test-verified=test_assigned.ftlh
+email-notifications.template-path.coach-assigned=test_assigned.ftlh
+email-notifications.template-path.coach-deassigned=test_assigned.ftlh
```
Comment (AntonMaksimenkau) [inline src/main/resources/application-release.properties]: can we set the correct template path's here?

File: src/main/resources/templates/test_assigned.ftlh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<html>
+<head>
+    <title>Test mail!</title>
+</head>
+<body>
+<h1>Hello, ${userName}!</h1>
+<p>Test is assigned for you</p>
```
Comment (AntonMaksimenkau) [inline src/main/resources/templates/test_assigned.ftlh]: Seams like a test value. Are you going to move creating templates to a separate task?

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/test_assigned.ftlh (hunk 4 above)
Comment (maxim12321) [reply src/main/resources/templates/test_assigned.ftlh]: I'm going to create templates for each notification type a bit later :)

**Review by AntonMaksimenkau — APPROVED** [review]

File: src/main/resources/application-release.properties [hunk 5]
```diff
@@ -10,12 +10,11 @@ cloud.aws.region.static=eu-central-1
...
 email-notifications.template-path.test-assigned=test_assigned.ftlh
-email-notifications.template-path.test-deassigned=test_assigned.ftlh
-email-notifications.template-path.test-started=test_assigned.ftlh
-email-notifications.template-path.test-verified=test_assigned.ftlh
-email-notifications.template-path.coach-assigned=test_assigned.ftlh
-email-notifications.template-path.coach-deassigned=test_assigned.ftlh
+email-notifications.template-path.test-deassigned=test_deassigned.ftlh
+email-notifications.template-path.test-verified=test_verified.ftlh
+email-notifications.template-path.coach-assigned=coach_assigned.ftlh
+email-notifications.template-path.coach-deassigned=coach_deassigned.ftlh
```
Comment (AntonMaksimenkau) [inline src/main/resources/application-release.properties:20]: Let's also add the template for expired tests

**Review by AntonMaksimenkau — APPROVED** [review]

File: src/main/resources/application-release.properties (hunk 5 above)
Comment (maxim12321) [reply src/main/resources/application-release.properties:20]: Oh, here we go again

**Review by Anlord241 — APPROVED** [review]
\+ 4.5k lines 😮

**Comment by tensenrey** [comment]
hacker :D

**Pull request merged**